compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -684,7 +685,14 @@ object ProtoTypes {
 
   /** Dummy tree to be used as an argument of a FunProto or ViewProto type */
   object dummyTreeOfType {
-    def apply(tp: Type)(implicit src: SourceFile): Tree = untpd.Literal(Constant(null)) withTypeUnchecked tp
+    /*
+     * A property indicating that the given tree was created with dummyTreeOfType.
+     * It is sometimes necessary to detect the dummy trees to avoid unwanted readaptations on them.
+     */
+    val IsDummyTree = new Property.Key[Unit]
+
+    def apply(tp: Type)(implicit src: SourceFile): Tree =
+      (untpd.Literal(Constant(null)) withTypeUnchecked tp).withAttachment(IsDummyTree, ())
```

Comment:
Maybe we should introduce a proper internal construct for these trees. Maybe have a `scala.internal.bla.dummyTree[T]` .

Reply:
Same for the `unapply`

Reply:
We can improve on this later.

Reply:
@nicolasstucki can you elaborate on what `scala.internal.bla.dummyTree[T]` should be? Some sort of special fictional value? I think creating a special `Tree` subclass for this purpose would work.

Reply:
@smarter it turned out that it was `dummyTreeOfType` that was the source of the `null` we talked about. Could you chime in with an opinion on whether it ever makes sense to `adapt` dummy trees created by this function?

Reply:
honestly no clue, I myself only learned about dummyTreeOfType recently.